ext/curl/interface.c

```diff
@@ -2086,6 +2100,78 @@ static inline int build_mime_structure_from_hash(php_curl *ch, zval *zpostfields
 			continue;
 		}
 
+		if (Z_TYPE_P(current) == IS_OBJECT && instanceof_function(Z_OBJCE_P(current), curl_CURLStringFile_class)) {
+			/* new-style file upload from string */
+			zval *prop, rv;
+			char *type = NULL, *filename = NULL;
+
+			prop = zend_read_property(curl_CURLStringFile_class, Z_OBJ_P(current), "postname", sizeof("postname")-1, 0, &rv);
+			ZVAL_DEREF(prop);
+			if (Z_TYPE_P(prop) != IS_STRING) {
```

Comment:
As `CURLStringFile::$postname` is a property of type `string`, is this condition true only if the property is uninitialized, right? If this is the case, shouldn't we throw an exception that we do in other cases? Or am I missing something?

Reply:
You're right, no need to call php_error_docref here. The exception is already being thrown inside zend_read_property. But I'm do not know how to process it correctly. Is it correct to do just this? : 
```
prop = zend_read_property(curl_CURLStringFile_class, Z_OBJ_P(current), "postname", sizeof("postname")-1, 0, &rv);
ZVAL_DEREF(prop);
if (Z_TYPE_P(prop) != IS_STRING) {
	zend_string_release_ex(string_key, 0);
	return FAILURE;
}
```
I cannot find a good example in other code =/

Reply:
That would be fine, or possibly something like this?

```
prop = ...;
if (EG(exception)) {
    zend_string_release_ex(string_key, 0);
    return FAILURE;
}
ZVAL_DEREF(prop);
ZEND_ASSERT(Z_TYPE_P(prop) == IS_STRING);
```